src/ast/nodes/MemberExpression.ts

```diff
@@ -9,13 +9,22 @@ import Identifier, { isIdentifier } from './Identifier';
 import NamespaceVariable from '../variables/NamespaceVariable';
 import ExternalVariable from '../variables/ExternalVariable';
 import { ForEachReturnExpressionCallback, SomeReturnExpressionCallback } from './shared/Expression';
-import { NodeType } from './NodeType';
+import * as NodeType from './NodeType';
 import { NodeRenderOptions, RenderOptions } from '../../utils/renderHelpers';
 import { isUnknownKey, ObjectPath, ObjectPathKey, UNKNOWN_KEY } from '../values';
 import { BLANK } from '../../utils/blank';
+import MetaProperty, { isMetaProperty } from './MetaProperty';
 
 const validProp = /^[a-zA-Z_$][a-zA-Z_$0-9]*$/;
 
+const globalImportMetaUrlMechanism = `(typeof document !== 'undefined' ? document.currentScript && document.currentScript.src || location.href : new URL('file:' + __filename).href)`;
+const importMetaUrlMechanisms: Record<string, string> = {
+	amd: `new URL(module.uri.startsWith('file:') ? module.uri : 'file:' + module.uri).href`,
+	cjs: `new (typeof URL !== 'undefined' ? URL : require('url').URL)('file:' + __filename).href`,
+	iife: globalImportMetaUrlMechanism,
+	umd: globalImportMetaUrlMechanism
+};
```

Comment:
My assumption was that it should be possible to switch an AMD module for a UMD modules without changed functionality which does not seem to be the case here. Maybe it is possible to make these two cases more similar in case an AMD loader is used?